assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -1,221 +1,161 @@
+			event.stopPropagation();
+			const context = getContext();
+			selectImage( context, 'next' );
+		},
+		selectPreviousImage: ( event: MouseEvent ) => {
+			event.stopPropagation();
+			const context = getContext();
+			selectImage( context, 'previous' );
+		},
+	},
+	callbacks: {
+		watchForChangesOnAddToCartForm: () => {
+			const context = getContext();
+			const variableProductCartForm = document.querySelector(
+				`form[data-product_id="${ context.productId }"]`
+			);
+
+			if ( ! variableProductCartForm ) {
+				return;
+			}
+
+			// TODO: Replace with an interactive block that calls `actions.selectImage`.
+			const observer = new MutationObserver( function ( mutations ) {
+				for ( const mutation of mutations ) {
+					const mutationTarget = mutation.target as HTMLElement;
+					const currentImageAttribute =
+						mutationTarget.getAttribute( 'current-image' );
+					if (
+						mutation.type === 'attributes' &&
+						currentImageAttribute &&
+						context.visibleImagesIds.includes(
+							currentImageAttribute
+						)
+					) {
+						context.selectedImage = currentImageAttribute;
+					}
 				}
+			} );
 
-				const observer = new MutationObserver( function ( mutations ) {
-					for ( const mutation of mutations ) {
-						const mutationTarget = mutation.target as HTMLElement;
-						const currentImageAttribute =
-							mutationTarget.getAttribute( 'current-image' );
-						if (
-							mutation.type === 'attributes' &&
-							currentImageAttribute &&
-							store.context.woocommerce.visibleImagesIds.includes(
-								currentImageAttribute
-							)
-						) {
-							store.context.woocommerce.selectedImage =
-								currentImageAttribute;
-						}
-					}
-				} );
+			observer.observe( variableProductCartForm, {
+				attributes: true,
+			} );
 
-				observer.observe( variableProductCartForm, {
-					attributes: true,
-				} );
+			return () => {
+				observer.disconnect();
+			};
+		},
+		keyboardAccess: () => {
+			const context = getContext();
+			let allowNavigation = true;
 
-				return () => {
-					observer.disconnect();
-				};
-			},
-			keyboardAccess: ( store: Store ) => {
-				const { context, actions } = store;
-				let allowNavigation = true;
+			const handleKeyEvents = ( event: Event ) => {
+				if ( ! allowNavigation || ! context.isDialogOpen ) {
+					return;
+				}
 
-				const handleKeyEvents = ( event: Event ) => {
-					if (
-						! allowNavigation ||
-						! context.woocommerce?.isDialogOpen
-					) {
-						return;
-					}
+				// Disable navigation for a brief period to prevent spamming.
+				allowNavigation = false;
 
-					// Disable navigation for a brief period to prevent spamming.
-					allowNavigation = false;
+				requestAnimationFrame( () => {
+					allowNavigation = true;
+				} );
 
-					requestAnimationFrame( () => {
-						allowNavigation = true;
-					} );
+				// Check if the esc key is pressed.
+				if ( event.keyCode === Keys.ESC ) {
```

Comment:
As mentioned above, `keyCode` is deprecated. It'd be worth considering replacing this (and other `event.keyCode` calls).

Reply:
@DAreRodz - thanks for the suggestions. I will do a follow up PR after merge.

Reply:
Wasn't too much work so I just pushed up the changes.